### ISSUE TITLE
Utiliser Next.js pour le site publicodes

### ISSUE THREAD

**Issue by mquandalle** [betagouv/mon-entreprise#1318]
Next.js a beaucoup gagné en popularité et permet de gérer des fonctionnalités avancées (server-side rendering, code splitting par page — utile pour ne pas charger Monaco dès la page d’accueil, etc.) sans avoir à configurer WebPack manuellement.

J'avais déjà envisagé la possibilité d'utiliser Next.js pour mon-entreprise.fr mais nous avons quelques parties du code qui seraient difficile à porter (systèmes de traductions et d'iframe notamment).

En revanche le site publi.codes est bien plus simple et sera bientôt dans son propre dépôt (il faudra dissocier sa configuration de "build" de celle de mon-entreprise), il me semble souhaitable et simple à mettre en œuvre d'utiliser Next.js pour ce site. Cela résoudrait #1282 par la même occasion.

**Comment by johangirod** [comment]
Yes, tout à fait d'accord. Mais une question, pourquoi next.js et pas gatsby ? C'est plus adapté il me semble et ça permettrait de ne pas trop faire monter la note d'hébergement...

**Comment by laem** [comment]
Je suis assez sceptique au sujet du SSR (mais vous le savez ^^). 

Pour nosgestesclimat.fr j'ai viré tout le code SSR de mon-entreprise pour simplifier la 1ère version en ligne, et on ne nous a remonté aucun problème de perf côté ressenti utilisateur. Le SEO n'a pas l'air d'en souffrir non plus. Ça ne montre pas que le SSR est inutile, mais plutôt qu'on peut vivre sans. Pour publicodes, je pense que l'intérêt n'est pas évident, n'étant pas un site grand public chargé 100k fois par jour.

Au sujet de next.js spécifiquement, voilà mon avis : quand on connait déjà bien React, webpack, j'ai peur qu'on ne gagne pas tant que ça de temps par rapport aux limitations potentielles. 
Cela dit, ce n'est qu'une intuition, et je vais bientôt me lancer dans un site NextJS donc je pourrai réellement comparer.
